oasis/models/rules.py

```diff
@@ -2,15 +2,151 @@
 
 from __future__ import absolute_import
 
-from oasis.models.model import Model
+import datetime
+import time
+from threading import Thread, Timer
+from pytimeparse.timeparse import timeparse
+from oasis.models.model import Model, Config
 from oasis.datasource import PrometheusAPI, PrometheusQuery, Metrics
+from oasis.libs.log import logger
+from oasis.models.util import sub_id, EPS
+from oasis.libs.features import Features
+from oasis.libs.alert import send_to_slack
+from oasis.libs.rules import parse_rule, OPERATORS
 
 
 class Rules(Model):
-    def __init__(self, job, callback):
+    def __init__(self, job):
+        super(Rules, self).__init__()
         self.job = job
-        self.callback = callback
         self.api = PrometheusAPI(job.data_source)
+        self.timer = Timer(timeparse(self.job.timeout), self.timeout_action)
+        self.config_file = "%s/rules.yml" % self.model_path
+        self.cfg = RulesConfig(self.config_file, job.config)
 
+    def run(self):
+        self.timer.start()
+        for key in self.job.metrics:
+            if key in Metrics:
+                val = Metrics[key]
+                if key not in self.cfg.metrics:
+                    continue
 
+                t = Thread(target=self.run_action,
+                           args=(key, val, self.cfg.metrics[key],
+                                 self.cfg.rules[key]))
+                t.start()
+                self.threads[key] = t
+
+    def run_action(self, metric, val, config, rules):
+        logger.info("[job-id:{id}][metric:{metric}] start ot run"
+                    .format(id=sub_id(self.job.id), metric=metric))
+        self.compute(metric, val, config, rules)
+
+    def compute(self, metric, query_expr, config, rules):
```

Comment:
add a comment for compute, I am confused about what data it want to confused, and return what data